website/source/docs/guides/kubernetes-reference.html.md

```diff
@@ -0,0 +1,184 @@
+allow you to have a fixed disk location for the Consul data. This ensures that
+if a Consul server is lost, the data will not be lost. This is an important
+feature of Kubernetes, but may take some additional configuration. If you are
+running Kubernetes on one of the major cloud platforms, persistent volumes
+should already be configured; be sure to read their documentation for more
+details. In addition to setting up the PV resource in Kubernetes, you will need
+to map the Consul server to that volume with the [storage class
+parameter](https://www.consul.io/docs/platform/k8s/helm.html#v-server-storageclass).
+
+Finally, you will need to enable RBAC on your Kubernetes cluster. Review
+[Kubernetes
+RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/),
+[AWS](https://docs.aws.amazon.com/eks/latest/userguide/managing-auth.html),
+[GCP](https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control),
+and
+[Azure](https://docs.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-create).
+In Azure, RBAC is enabled by default . 
+
+## Datacenter Design 
+
+There are many configurations for running Consul with Kubernetes. In this guide
+we will cover three of the most common.
+
+1. Consul agents can be solely deployed within kubernetes.  
+1. Consul servers
+can be deployed outside of kubernetes and clients inside of Kubernetes.  
+1. Multiple Consul datacenters with agents inside and outside of Kubernetes.  
+
+Review the Kubernetes
+[documentation](https://www.consul.io/docs/platform/k8s/index.html#use-cases)
+for additional use case information. 
+
+Since all three use cases will also need catalog sync, review the
+implementation details for catalog sync.  
+
+### Consul Datacenter Deployed in Kubernetes 
+
+Deploying a Consul cluster, servers and clients, in Kubernetes can be done with
+the official [Helm
+chart](https://www.consul.io/docs/platform/k8s/helm.html#using-the-helm-chart).
+This configuration is useful for managing services within Kubernetes and is
+common for users who do not already have a production Consul datacenter.
+
+![Reference Diagram](/assets/images/k8s-consul-simple.png "Consul in Kubernetes Reference Diagram")
+
+The Consul datacenter in Kubernetes will function the same as a platform
+independent Consul datacenter. Agents will communicate over LAN Gossip, servers
+will participate in the Raft consensus and quorum, and client requests will be
+forwarded to the leading server with the RPC.
+
+### Consul Datacenter with a Kubernetes Cluster
+
+To use an existing Consul cluster to manage services in Kubernetes, Consul
+clients can be deployed within the Kubernetes cluster. This will also allow the
+Kubernetes services to be synced to Consul. This design allows Consul tools
+such as envconsul, consul-template, and more to work on Kubernetes. It will
+also register each Kubernetes node with the Consul catalog for full visibility
+into your infrastructure.
```

Comment:
Catalog sync doesn't register Kube nodes. Suggest we remove the last sentence. Agent's register themselves the normal way.

Reply:
Agree, we don't directly register Kubernetes nodes. We have some visibility into them because clients are deployed as a daemonset (one per Kube node), but it's a side effect.

Reply:
Sounds good! I will remove the last sentence.